### ISSUE TITLE
Collections with string as key not possible

### ISSUE THREAD

**Issue by mrk-j** [nunomaduro/larastan#1133]
- Larastan Version: 2.0.1
- `--level` used: 5

### Description
When I try to use a `string` as the key for a value in a `Collection` the following error is found:

```
 ------ ---------------------------------------------------------------------------------------------------------
  Line   Providers/AppServiceProvider.php
 ------ ---------------------------------------------------------------------------------------------------------
  29     Parameter #1 $key of method Illuminate\Support\Collection<int,mixed>::put() expects int, string given.
  30     Parameter #1 $key of method Illuminate\Support\Collection<int,mixed>::pull() expects int, string given.
 ------ ---------------------------------------------------------------------------------------------------------
```

It seems I cannot override the key type for the `Collection`?

### Laravel code where the issue was found

```php
$collection = collect();
$collection->put('foo', 'bar');
$collection->pull('foo');
```

**Comment by canvural** [comment]
Hi,

Yeah it should be benevolent union of integer and string type [here](https://github.com/nunomaduro/larastan/blob/master/src/ReturnTypes/Helpers/CollectExtension.php#L41) instead of just IntegerType.

**Comment by szepeviktor** [comment]
> But I guess the `MixedType` is not valid here since objects and arrays cannot be used as keys?

Here you go! https://www.php.net/manual/en/language.types.array.php#example-54

**Comment by canvural** [comment]
> Setting the key type to `MixedType` in the line mentioned above fixes the error for me. But I guess the `MixedType` is not valid here since objects and arrays cannot be used as keys?

I already wrote the correct type above. It shouldn't be `MixedType`

**Comment by canvural** [comment]
Fixed: b0a50dfe2d21c44c9f2577aa5d7c9825b6b9d5b2
